File: java/review/R1710.java
package review;

public class R1710 {
    public int majorityElement(int[] nums) {
        int res = 0;
        int cnt = 0;
        for (int num : nums) {
            if (cnt == 0) {
                res = num;
                ++cnt;
            } else if (num == res) {
                ++cnt;
            } else {
                --cnt;
            }
        }
        cnt = 0;
        for (int num : nums) {
            if (num == res) {
                ++cnt;
            }
        }
        return cnt > nums.length / 2 ? res : -1;
    }
}

File: cpp/review/r1710.cc
#include <vector>
using std::vector;
#include <iostream>

class Solution {
public:
    vector<int> missingTwo(vector<int>& nums) {
        std::size_t nz{nums.size()};
        int a = 0, b = 0;
        for (std::size_t i{0};i < nz;++i) {
            int num = nums[i];
            if (num == 0 || num == i + 1) {
                continue;
            }
            nums[i] = 0;
            if (num > nz) {
                if (a != 0) {
                    b = num;
                } else {
                    a = num;
                }
            }
            while (num <= nz && num != 0 && num != nums[num - 1]) {
                int nid = num - 1;
                int nNum = nums[nid];
                nums[nid] = num;
                num = nNum;
            }
        }
        int j = 1;
        vector<int> res;
        for (;j <= nz;++j) {
            if (nums[j - 1] == 0) {
                res.push_back(j);
            }
        }
        while (res.size() < 2) {
            if (j == a || j == b) {
                ++j;
                continue;
            }
            res.push_back(j++);
        }
        return res;
    }
};